### PR TITLE
pygame.draw.circles() function

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
Implements one of #3269. Follows #3324.
This one implements the `circles()` function for drawing many circles at once. The function's use case should be heterogeneous circle sequences, with both hollow and filled ones of different scales and colors.
Unlike `draw.circle()`, each circle in the sequence is described by either 3 or 4 parameters, indicating color, center_pos, radius, width(optional).

First test show a 20-55% performance benefit over a raw for loop with `draw.circle()`.

### CODE REVIEW

**Comment by MyreMylar** [comment]
Just toggling this one closed and open again to rerun the tests as one run had expired. 

Possible it may need to merge with main again to pass. We shall see

File: buildconfig/stubs/pygame/draw.pyi
Comment (MyreMylar) [inline buildconfig/stubs/pygame/draw.pyi:38]: So one slightly annoying thing about this `Optional[int]` type hint is that it means "An integer value or `None`, is expected here". If you put only three parameters into the tuple you get a typing error, but it will still work.

Perhaps this hint should be:

`Sequence[Union[Tuple[ColorValue, Coordinate, int, int], Tuple[ColorValue, Coordinate, int]]]`

**Review by MyreMylar — CHANGES_REQUESTED** [review]
I used this test program:

```
import pygame
import random

pygame.init()

display_surface = pygame.display.set_mode((800, 600))
background = pygame.Surface((800, 600))
background.fill((158, 158, 170))
running = True

font = pygame.font.Font(None, 16)
mouse_pos_render = font.render("Mouse Pos: " + str(pygame.mouse.get_pos()), True, pygame.Color('#FFFFFF'))

clock = pygame.time.Clock()

circles_to_draw = [(pygame.Color(random.randint(50, 220),
                                 random.randint(50, 220),
                                 random.randint(50, 220)),
                   (random.randint(50, 750), random.randint(50, 550)),
                   random.randint(5, 50), 0) for _ in range(0, 4000)]

delta_times = []

while running:
    time_delta = max(clock.tick()/1000.0, 0.000000001)

    if len(delta_times) > 100:
        delta_times.pop(0)
    delta_times.append(time_delta)

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    display_surface.blit(background, (0, 0))

    for circle in circles_to_draw:
        pygame.draw.circle(display_surface, circle[0], circle[1], circle[2])

    #pygame.draw.circles(display_surface, circles_to_draw)

    mouse_pos_render = font.render("Mouse Pos: " + str(pygame.mouse.get_pos()), True, pygame.Color('#FFFFFF'))
    display_surface.blit(mouse_pos_render, (10, 10))

    mouse_pos_render = font.render("FPS: " + f"{(1.0 / (sum(delta_times)/len(delta_times))):.02f}", True, pygame.Color('#FFFFFF'))
    display_surface.blit(mouse_pos_render, (700, 10))

    pygame.display.flip()

```

Uncommenting to switch between the two methods of drawing 4,000 circles. 

In my testing using `circles()` was about 15% to 20% faster than using circle to do the same thing.

I think the API seems reasonable and fits with the pattern established by `draw.line()` & `draw.lines()`. I will suggest that users are less likely to draw a lot of circles all at once than they are to draw a lot of lines at once - but on balance I think I approve of this new API.

I would like the type hint fixed though.

File: buildconfig/stubs/pygame/draw.pyi
Comment (itzpr3d4t0r) [reply buildconfig/stubs/pygame/draw.pyi:38]: About the perf tests, i think we have the same issue here that we have with fblits or blits. Looping will be more efficient with smaller circles or hollow ones so performance gains can vary significantly.

Comment (itzpr3d4t0r) [reply buildconfig/stubs/pygame/draw.pyi:38]: btw i still have to add many tests to this PR and all my other PRs in #3269. Might discover more issues when i do so. Just waiting for reviews like this to make sure i don't spend time where it's not wanted/needed.

**Pull request closed without merging**
